satpy/composites/__init__.py

```diff
@@ -153,14 +153,46 @@ def apply_modifier_info(self, origin, destination):
                     d[k] = o[k]
 
     def match_data_arrays(self, data_arrays):
-        """Match data arrays so that they can be used together in a composite."""
+        """Match data arrays so that they can be used together in a composite.
+
+        For the purpose of this method, "can be used together" means:
+
+        - All arrays should have the same dimensions.
+        - Either all arrays should have an area, or none should.
+        - If all have an area, the areas should be all the same.
+
+        In addition, negligible non-dimensional coordinates are dropped (see
+        :meth:`drop_coordinates`) and dask chunks are unified (see
+        :func:`satpy.utils.unify_chunks`).
+
+        Args:
+            data_arrays (List[arrays]): Arrays to be checked
```

Comment:
I suppose we never made a decision on type annotations versus docstring types did we? @mraspaud @sfinkens @pnuu any opinions?

```python
def func(a: int, b: list[str]) -> list[int]:
    """Some docstring.

    Args:
        a: description
        b: description

    Returns:
        description
    """
```

versus only putting them in the docstring?

Reply:
I just tested locally.  If I put the types only in the annotations, then the sphinx documentation reports the argument types in the signature, but not in the documentation body.  The return type it documents in both locations.  Maybe this is configurable, but if omitted in the docstring I think is not an improvement.  Or would you put the information in both locations, duplicating it?

Reply:
There may be a sphinx extension we need to make it handle the arguments (I would hope it would come out of the box in the future). I'd prefer not duplicating.

Reply:
The logical place to put it is as type annotations of course, but sphinx not picking it up for the docstring is annoying of course. Can this help? https://github.com/tox-dev/sphinx-autodoc-typehints

Reply:
or this https://sphinx-toolbox.readthedocs.io/en/latest/extensions/more_autodoc/typehints.html

Reply:
Looking at xarray it looks like they tell "autodoc" to not include type hints and they have type annotations and docstring types in the source code. That's probably not what they want long term though and may just be a work around. Looks like they also have aliases for types/functions so they don't have to use long names in their docs when linking to things.

https://github.com/pydata/xarray/blob/main/doc/conf.py#L112

Reply:
So what is the conclusion here?  Should I change this to have the type only in annotations?

Reply:
Leave it as-is for now. It isn't how I would do it, but that definitely doesn't make it correct. We'll have to look at this in some other PR as far as the sphinx docs handling it correctly. If you are bored feel free to try one or more of them out and see how it renders if you only include type annotations and nothing in the docstring.

I'll merge this if there are no more comments.